animations-baker.js

```diff
@@ -344,10 +345,23 @@ const baker = async (uriPath = '', fbxFileNames, vpdFileNames, outFile) => {
           // const bone = bones.find(b => b.name === boneName);
           const boneInterpolant = new THREE.QuaternionLinearInterpolant(track.times, track.values, track.getValueSize());
           boneQuaternionInterpolants[boneName] = boneInterpolant;
+        } else if (/\.scale$/.test(track.name)) {
+          if (allOnes(track.values)) {
+            const index = tracks.indexOf(track);
+            if (index !== -1) {
+              tracksToRemove.push(index);
+            }
+          } else {
+            throw new Error(`Error with the following track.  All scale transforms must be set to 1. Aborting.\n Animation: ${animation.name}, Track: ${track.name}, values: \n ${track.values}`);
+          }
         } else {
           console.warn('unknown track name', animation.name, track);
         }
       }
+      // remove scale transform tracks as they won't be used;
+      tracksToRemove.forEach(i => {
```

Comment:
I don't think this is legal/correct for all combinations. Splicing changes the index that you must splice for other items.

Reply:
yes you're right, will adjust

Reply:
corrected to go backwards, to prevent changing index's for values which need to be removed